extensions/chromium/preserve-referer.js

```diff
@@ -60,9 +66,32 @@ function forgetReferrerEventually(tabId) {
   g_referrerTimer[tabId] = setTimeout(() => {
     delete g_referrers[tabId];
     delete g_referrerTimer[tabId];
+    delete g_postRequests[tabId];
   }, REFERRER_IN_MEMORY_TIME);
 }
 
+// Keeps track of whether a document in tabId + frameId is loaded through a
+// POST form submission. Although this logic has nothing to do with referrer
+// tracking, it is still here to enable re-use of the webRequest listener above.
+function setCanRequestBody(tabId, frameId, isPOST) {
+  if (isPOST) {
+    if (!g_postRequests[tabId]) {
```

Comment:
For such cases (there are other occurrences in a previous commit for this file I believe) we could use the `||=` operator to shorten the code; see https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Logical_OR_assignment

Reply:
While `||=` is the equivalent operator here, in this specific case the if-check was to check whether it was undefined vs `Set`. Thus I'm adapting your suggestion by using `??=` instead.